### ISSUE TITLE
about train 

### ISSUE THREAD

**Issue by LeiLu-ux** [ultralytics/ultralytics#16847]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How do I set the image to 640x512 for training

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LeiLu-ux, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com) for detailed guidance, where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many of the most common questions may already be answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist our debugging efforts.

For your specific ❓ Question about setting the image to 640x512 for training, it's best to delve into the configuration files and scripts. Providing more context such as the training scripts you're using can help us guide you better.

Join the Ultralytics community for discussions and support:
- [Discord](https://ultralytics.com/discord) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com) for in-depth discussions
- [Subreddit](https://reddit.com/r/ultralytics) for community knowledge sharing

## Upgrade

Ensure you have the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) by upgrading in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in updated environments (all dependencies included):
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this CI badge is green, all tests are currently passing, ensuring the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

This is an automated response, but an Ultralytics engineer will assist you soon. 😊

**Comment by Y-T-G** [comment]
You can only use a square size during training, or use `rect=True` which will resize the longest side to `imgsz` while maintaining the aspect ratio.

But you can export at non-square size.
